cmd/sync.go

```diff
@@ -477,36 +477,21 @@ Final Job Status: %v%s%s
 func (cca *cookedSyncCmdArgs) process() (err error) {
 	ctx := context.WithValue(context.TODO(), ste.ServiceAPIVersionOverride, ste.DefaultServiceApiVersion)
 
-	// verifies credential type and initializes credential info.
-	// For sync, only one side need credential.
-	cca.credentialInfo.CredentialType, err = getCredentialType(ctx, rawFromToInfo{
-		fromTo:         cca.fromTo,
-		source:         cca.source,
-		destination:    cca.destination,
-		sourceSAS:      cca.sourceSAS,
-		destinationSAS: cca.destinationSAS,
-	})
+	// Initializes credential info and oauth token for the destination. This should not be re-used on the source.
```

Comment:
I'm finding this confusing: in that it's named dst... but sometimes we put source credentials into it.

Is there anything that would stop us from defining both cca.srcCredentialInfo and cca.dstCredentialInfo, and always putting the source creds in the source one and the dest creds in the dest one?

Reply:
In fact, I think we really ought to define both.  Right now we are getting issues - e.g. I ran a copy based download, and the isSource flag was wrong, becuase of the way this stuff currently works.  Getting separate fields would help.